### ISSUE TITLE
Serious data loss: deleting file deletes other files' changes

### ISSUE THREAD

**Issue by danielchasehooper** [git-up/GitUp#899]
using gitup version 1.3.2 (1045)
macOS 13.1 (22C65)
steps to repro:

1. run these commands
```
mkdir bug
cd bug
echo '#include <stdio.h>
int main() {
    printf("hello\n");
}' > main.c
git init
git add main.c
git commit -m "initial commit"
cc -o main main.c
echo "// here are some changes to file" >> main.c
```

2. Open the `bug/` folder in gitup
3. go to gitup's commit view
4. select "main" in the `working directory` section
5. press delete on your keyboard

result: both `main` and `main.c` will lose all their changes, even though only `main` was selected

**Comment by danielchasehooper** [comment]
@swisspol @lucasderraugh This is pretty serious, I lost a lot of work because of it.

**Comment by lucasderraugh** [comment]
I'm going to revert back to 1.3.2 version of libgit2 today. Please check for an update later today or revert to 1.3.2 in the meantime.

**Comment by lucasderraugh** [comment]
Oh, this was on 1.3.2? Then I'll take some time to investigate your repro steps.

**Comment by lucasderraugh** [comment]
This was actually fixed as part of 1.3.4, so I'm going to close this particular ticket. 1.3.4 breaks some things but fixes others I guess...
